File: 2023-2024/sourcecode/mod1/lab2/22SelectionBasicsJava/app/src/main/java/com/example/a22_selectionbasics_java/NumberListAdapter.java
package com.example.a22_selectionbasics_java;

import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.selection.SelectionTracker;
import androidx.recyclerview.widget.RecyclerView;

import com.example.a22_selectionbasics_java.databinding.RvItemBinding;

import java.util.LinkedList;

public class NumberListAdapter extends RecyclerView.Adapter<NumberListViewHolder>{

    private final LinkedList<Integer> numberList;

    private SelectionTracker<Long> selectionTracker;
    public void setSelectionTracker(SelectionTracker<Long> selectionTracker) {
        this.selectionTracker = selectionTracker;
    }

    public NumberListAdapter(LinkedList<Integer> numberList){
        this.numberList = numberList;
        setHasStableIds(true);
    }

    @NonNull
    @Override
    public NumberListViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new NumberListViewHolder(RvItemBinding.inflate(
                LayoutInflater.from(parent.getContext()),
                parent,
                false
        ));
    }

    @Override
    public void onBindViewHolder(@NonNull NumberListViewHolder holder, int position) {
        holder.bind(
                numberList.get(position),
                selectionTracker.isSelected((long)position));
    }

    @Override
    public int getItemCount() {
        return numberList.size();
    }

    @Override
    public long getItemId(int position) {
        return position;
    }
}

File: 2023-2024/sourcecode/mod1/lab2/21SelectorBasicsKotlin/app/src/main/java/com/example/a21_selectorbasics_kotlin/NumberListAdapter.kt
package com.example.a21_selectorbasics_kotlin

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.selection.SelectionTracker
import androidx.recyclerview.widget.RecyclerView
import com.example.a21_selectorbasics_kotlin.databinding.RvItemBinding

class NumberListAdapter(
    private val numberList: List<Int>,
) : RecyclerView.Adapter<NumberListViewHolder>() {

    lateinit var selectionTracker: SelectionTracker<Long>

    init {
        setHasStableIds(true)
    }

    override fun onCreateViewHolder(
        parent: ViewGroup,
        viewType: Int
    ): NumberListViewHolder {
        return NumberListViewHolder(
            RvItemBinding.inflate(
                LayoutInflater.from(parent.context), parent, false
            ))
    }

    override fun onBindViewHolder(holder: NumberListViewHolder, position: Int) {
        holder.bind(numberList[position], selectionTracker.isSelected(position.toLong()))
    }

    override fun getItemCount() = numberList.size
    override fun getItemId(position: Int): Long = position.toLong()
}